base/bundle.cpp

```diff
@@ -0,0 +1,113 @@
+#pragma once
+
+#include "base/bundle.hpp"
+
+#include <functional>
+#include <queue>
+
+#include "base/map_util.hpp"
+#include "base/status.hpp"
+
+namespace principia {
+namespace base {
+
+Bundle::Bundle(int threads) {
+  {
+    std::unique_lock<std::mutex> l(lock_);
+    joining_ = false;
+  }
+  for (int i = 0; i < threads; ++i) {
+    threads_.emplace_back([this]() {
+      std::shared_ptr<Task> current_task;
```

Comment:
Please extract this lambda to a member function.  It is hard to figure out what the constructor is doing, especially with the lambdas nested within the lambda.